config/action.d/pbr.conf

```diff
@@ -0,0 +1,66 @@
+# Fail2Ban configuration file
+#
+# Author: Konstantin Shalygin <k0ste@k0ste.ru>
+#
+# Policy Based Routing rules
+#
+# Provides the same features, like 'route' action, and fixes the main issue of
+# this hardening blocking: block per ip + service port + service protocol.
+# Actually you can add more options and do more granular blocking, see `ip rule
+# add help`
+
+[Definition]
+actionstart = <pbr_<type>/actionstart>
+actionstop = <pbr_<type>/actionstop>
+actionban = <pbr_<type>/actionban>
+actionunban = <pbr_<type>/actionunban>
+_matchers = <pbr_<type>/match>
+
+[pbr_simple]
+actionstart =
+actionstop =
+actionban = ip route add <blocktype> <_matchers>
+actionunban = ip route del <blocktype> <_matchers>
+match = <ip>
+
+[pbr_allports]
+actionstart = ip route add <blocktype> 0.0.0.0/0 table <tableid>
+actionstop = ip route delete <blocktype> 0.0.0.0/0 table <tableid>
+actionban = ip rule add to <ip> <_matchers> lookup <tableid>
+actionunban = ip rule delete to <ip> <_matchers> lookup <tableid>
+match = ipproto <protocol>
+
+[pbr_multiport]
+actionstart = <pbr_allports/actionstart>
+actionstop = <pbr_allports/actionstop>
+actionban = <pbr_allports/actionban>
+actionunban = <pbr_allports/actionunban>
+match = <pbr_allports/match> sport <port>
```

Comment:
Does not `sport` means source port? And would it need destination port `dport` by the rule instead?

Reply:
Why did it resolved? The question is unanswered yet... and my guess it must be dport instead of sport, since by incomming traffic the dport means the listener port of the service and sport - the opposite side (means port on source system connecting to the listener). I doubt it is correct.